text/0605-eks-rewrite.md

```diff
@@ -0,0 +1,476 @@
+  kubectlMemory: Size.gigabytes(4),
+  kubectlLayer: new KubectlV31Layer(this, 'kubectl'),
+  kubectlEnvironment: {
+    'http_proxy': 'http://proxy.myproxy.com',
+  },
+  kubectlRole: iam.Role.fromRoleArn(
+    this,
+    'MyRole',
+    'arn:aws:iam::123456789012:role/lambda-role'
+  );
+});
+```
+
+Usage - After
+
+```
+new eks.Cluster(this, 'MyCluster', {
+  version: eks.KubernetesVersion.V1_31,
+  kubectlLayer: new KubectlV31Layer(this, 'kubectl'),
+  kubectlProviderOptions: {
+    memory: Size.gigabytes(4),
+    environment: {
+      'http_proxy': 'http://proxy.myproxy.com',
+    },
+    role: iam.Role.fromRoleArn(
+      this,
+      'MyRole',
+      'arn:aws:iam::123456789012:role/lambda-role'
+    );
+});
+```
+
+## Migration Path
+
+Note: We can't guarantee it's a safe migration.
+
+Due to the fact that switching from a custom resource
+(Custom::AWSCDK-EKS-Cluster) to a native L1 (AWS::EKS::Cluster) resource
+requires cluster replacement, CDK users who need to preserve their cluster will
+have to take additional actions.
+
+1. Set the authentication mode of cluster from `AuthenticationMode.CONFIG_MAP`
+   to `AuthenticationMode.API_AND_CONFIG_MAP` and deploy
+2. Set the authentication mode of cluster from
+   `AuthenticationMode.API_AND_CONFIG_MAP` to `AuthenticationMode.API` and
+   deploy
+3. Set removal policy to RETAIN on the existing cluster (and manifests) and
+   deploy.
+4. Remove cluster definition from their CDK app and deploy
+5. Add new cluster definition using the new constructs(EKSV2).
+6. Follow cdk import to import the existing cluster as the new definition.
+   1. All relevant EKS resources support import.
+   2. AWS::EKS::Cluster
+   3. AWS::EKS::FargateProfile
+   4. AWS::EKS::Nodegroup
+7. Add Manifests.
+
+## Public FAQ
+
+### What are we launching today?
+
+We’re launching a new EKS module `aws-eksv2-alpha`. It's a rewrite of existing
+`aws-eks` module with some breaking changes based on community feedbacks.
+
+### Why should I use this feature?
+
+The new EKS module provides faster deployment, less complexity, less cost and
+more features (e.g. isolated VPC and escape hatching).
+
+### What's the future plan for existing `aws-eks` module?
+
+- When the new alpha module is published, `aws-eks` module will enter
+  `maintenance` mode which means we will only work on bugs on `aws-eks` module.
+  New features will only be added to the new `aws-eksv2-alpha` module. (Note:
+  this is the general guideline and we might be flexible on this)
+- When the new alpha module is stabilized, `aws-eks` module will transition into
+  a deprecation phase. This implies that customers should plan to migrate their
+  workloads to the new module. While they can continue using the old module for
+  the time being, CDK team will prioritize new features/bug fix on the new
+  module
+
+## Internal FAQ
+
+### Why are we doing this?
+
+This feature has been highly requested by the community since Feb 2023. The
```

Comment:
Show evidence. Lets also put the list of issues that are going to get resolved with this change.

Reply:
Added a list of Github issues that will be resolved with the new module.